cmd/bosun/sched/check.go

```diff
@@ -322,22 +324,38 @@ func (s *Schedule) CollectStates() {
 func (r *RunHistory) GetUnknownAndUnevaluatedAlertKeys(alert string) (unknown, uneval []expr.AlertKey) {
 	unknown = []expr.AlertKey{}
 	uneval = []expr.AlertKey{}
+	anyFound := false
 	for ak, ev := range r.Events {
 		if ak.Name() != alert {
 			continue
 		}
+		anyFound = true
 		if ev.Status == StUnknown {
 			unknown = append(unknown, ak)
 		} else if ev.Unevaluated {
 			uneval = append(uneval, ak)
 		}
 	}
+	if !anyFound {
```

Comment:
That is correct. This allows dependencies to work even if the dependent-upon alert has not been run this interval. If the required alert was not run this interval, we simply look at the current state from the last run. We can discuss tomorrow.